api/handlers.go

```diff
@@ -743,24 +743,25 @@ func postImagesCreate(c *context, w http.ResponseWriter, r *http.Request) {
 		errorFound := false
 		nonOSErrorFound := false
 		successfulPull := false
-		callback := func(what, status string, err error) {
-			if err != nil {
+		callback := func(msg cluster.JSONMessageWrapper) {
+			msg.Msg.ID = msg.EngineName
+			if msg.Msg.Status != "" {
+				msg.Msg.Status = fmt.Sprintf("Pulling %s... : %s", image, msg.Msg.Status)
+			} else {
+				msg.Msg.Status = fmt.Sprintf("Pulling %s...", image)
+			}
+			if msg.Success {
+				successfulPull = true
+			}
```

Comment:
If you intention is `successfulPull = msg.Success`, you should change it. Otherwise maybe good to add a comment on corner cases.